aws/resource_aws_kinesis_stream_consumer.go

```diff
@@ -0,0 +1,229 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsKinesisStreamConsumer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsKinesisStreamConsumerCreate,
+		Read:   resourceAwsKinesisStreamConsumerRead,
+		Delete: resourceAwsKinesisStreamConsumerDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(120 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
```

Comment:
since the `DescribeStreamConsumer` and `Deregister` API methods support searching by ARN, we can set the ID with determined ARN for easier lookup throughout 👍